pkg/nodeagent/pod_controller.go

```diff
@@ -237,101 +242,122 @@ func buildRmdWorkload(pod *corev1.Pod) ([]*intelv1alpha1.RmdWorkload, error) {
-				if err != nil {
-					continue
-				}
-				rmdWorkload.Spec.Rdt.Cache.Min = minCache
-			case strings.HasSuffix(field, mbaPercentageConst):
-				mbaPercentage, err := strconv.Atoi(data)
-				if err != nil {
-					continue
-				}
-				rmdWorkload.Spec.Rdt.Mba.Percentage = mbaPercentage
-			case strings.HasSuffix(field, mbaMbpsConst):
-				mbaMbps, err := strconv.Atoi(data)
-				if err != nil {
-					continue
-				}
-				rmdWorkload.Spec.Rdt.Mba.Mbps = mbaMbps
-			case strings.HasSuffix(field, pstateRatioConst):
-				if data != "" {
-					rmdWorkload.Spec.Plugins.Pstate.Ratio = data
-				}
-			case strings.HasSuffix(field, pstateMonitoringConst):
-				if data != "" {
-					rmdWorkload.Spec.Plugins.Pstate.Monitoring = data
-				}
-			}
-
-		}
 		rmdWorkloads = append(rmdWorkloads, rmdWorkload)
 	}
 	return rmdWorkloads, nil
 }
 
+func getAnnotationInfo(rmdWorkload *intelv1alpha1.RmdWorkload, pod *corev1.Pod, containerName string) {
+	var minCacheStr, mbaPercentStr, mbaMbpsStr string = "0", "0", "0" //Ensures strconv.Atoi() doesn't receive an empty string
+	workloadData := pod.GetObjectMeta().GetAnnotations()
+	for field, data := range workloadData {
+		if !strings.HasPrefix(field, containerName) {
+			continue
+		}
+		switch {
+		case strings.HasSuffix(field, policyConst):
+			if data != "" {
+				rmdWorkload.Spec.Policy = data
+			}
+		case strings.HasSuffix(field, cacheMinConst):
+			if data != "" {
+				minCacheStr = data
+			}
+		case strings.HasSuffix(field, mbaPercentageConst):
+			if data != "" {
+				mbaPercentStr = data
+			}
+		case strings.HasSuffix(field, mbaMbpsConst):
+			if data != "" {
+				mbaMbpsStr = data
+			}
+		case strings.HasSuffix(field, pstateRatioConst):
+			if data != "" {
+				rmdWorkload.Spec.Plugins.Pstate.Ratio = data
+			}
+		case strings.HasSuffix(field, pstateMonitoringConst):
+			if data != "" {
+				rmdWorkload.Spec.Plugins.Pstate.Monitoring = data
+			}
+		}
+	}
+	var err error
+	rmdWorkload.Spec.Rdt.Cache.Min, err = strconv.Atoi(minCacheStr)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	rmdWorkload.Spec.Rdt.Mba.Percentage, err = strconv.Atoi(mbaPercentStr)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	rmdWorkload.Spec.Rdt.Mba.Mbps, err = strconv.Atoi(mbaMbpsStr)
+	if err != nil {
+		fmt.Println(err)
+	}
+}
+
+func getContainerInfo(pod *corev1.Pod, container corev1.Container) (containerInformation, error) {
+	var containerInfo containerInformation //empty containerInformation struct
+
+	logger := log.WithName("buildRmdWorkload")
+	if !exclusiveCPUs(pod, &container) {
+		logger.Info("No container requesting cache found in pod")
+		return containerInformation{}, nil
+	}
+	podUID := string(pod.GetObjectMeta().GetUID())
+	if podUID == "" {
+		logger.Info("No pod UID found")
+		return containerInformation{}, errors.NewServiceUnavailable("pod UID not found")
+	}
+
+	containerID := getContainerID(pod, container.Name)
+	coreIDs, err := readCgroupCpuset(podUID, containerID)
+	containerInfo.coreIDs = coreIDs
```

Comment:
We should wait to assign coreIDs until after the error check and also after `if len(coreIDs) == 0 {...` check.